### PR TITLE
[explorer]: update gas payment section

### PR DESCRIPTION
**Pull request by plam-ml**
## Description 

- Update gas payment section so it will always display at least 2 items
- When there are over 10 items, it will become scollable
- https://mysten.atlassian.net/browse/APPS-460
- DEMO: https://explorer-66hwh2ogf-mysten-labs.vercel.app/txblock/J4zaDNsMTAMrpFe5epDPzRwVR9TwNmLQx8mzKBvxvkA9?network=testnet

Screenshot 2023-05-10 at 4 18 27 PM

## Test Plan 

How did you test the new or updated feature?

---
If your changes are not user-facing and not a breaking change, you can skip the following section. Otherwise, please indicate what changed, and then add to the Release Notes section as highlighted during the release process.

### Type of Change (Check all that apply)

- [ ] user-visible impact
- [ ] breaking change for a client SDKs
- [ ] breaking change for FNs (FN binary must upgrade)
- [ ] breaking change for validators or node operators (must upgrade binaries)
- [ ] breaking change for on-chain data layout
- [ ] necessitate either a data wipe or data migration

### CODE REVIEW

File: apps/explorer/src/components/GasBreakdown/index.tsx [hunk 1]
```diff
@@ -91,8 +94,15 @@ function TotalGasAmount({ amount }: GasProps) {
...
 
 function GasPaymentLinks({ objectIds }: { objectIds: string[] }) {
+    const isScrollable = objectIds.length > MAX_GAS_PAYMENT_LENGTH;
+
     return (
-        <div className="flex flex-wrap items-center gap-x-4">
+        <div
+            className={clsx(
+                'flex flex-wrap items-center gap-x-4 gap-y-2',
+                isScrollable && 'h-20 overflow-y-scroll'
```
Comment (Jordan-Mysten) [inline apps/explorer/src/components/GasBreakdown/index.tsx:103]: Why not just do max-h-20 and overflow scroll? Then you wouldn't need this logic?

Comment (petvaizAkhtar) [reply apps/explorer/src/components/GasBreakdown/index.tsx:103]: Reply

File: apps/explorer/src/pages/transaction-result/TransactionData.tsx [hunk 2]
```diff
@@ -70,7 +70,7 @@ export function TransactionData({ transaction }: Props) {
                 )}
             </section>
 
-            <section className="flex w-96 flex-1 flex-col gap-6">
+            <section className="flex w-96 flex-1 flex-col gap-6 md:min-w-[510px]">
```
[the remark this answers is not in the document]
Comment (petvaizAkhtar) [reply apps/explorer/src/pages/transaction-result/TransactionData.tsx:73]: Please reply

**Review by mystie711 — APPROVED** [review]
lookin good.

**Pull request merged**